docs/using-hive.md

```diff
@@ -104,15 +104,41 @@ spec:
     name: mycluster-ssh-key
 ```
 
-The hiveutil CLI (see `make hiveutil`) offers a create-cluster command for generating a cluster deployment and submitting it to the cluster defined by your current kubeconfig.
+The hiveutil CLI (see `make hiveutil`) offers a create-cluster command for generating a cluster deployment and submitting it to the Hive cluster using your current kubeconfig.
+
+To view what create-cluster generates, *without* submitting it to the API server, add `-o yaml` to the above command. If you need to make any changes not supported by create-cluster options, the output can be saved, edited, and then submitted with `oc apply`.
+
+By default this command assumes the latest Hive master CI build, and the latest OpenShift stable release. `--hive-image` can be specified to use a specific Hive image to run the install, and `--release-image` can be specified to control which OpenShift release image to install in the cluster.
+
+#### Create Cluster on AWS
+
+Credentials will be read from your AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY environment variables. Alternatively you can specify an AWS credentials file with --creds-file.
 
 ```bash
-bin/hiveutil create-cluster --base-domain=mydomain.example.com mycluster
+bin/hiveutil create-cluster --base-domain=mydomain.example.com --cloud=aws mycluster
 ```
 
-To view what create-cluster generates, *without* submitting it to the API server, add `-o yaml` to the above command. If you need to make any changes not supported by create-cluster options, the output can be saved, edited, and then submitted with `oc apply`.
+#### Create Cluster on Azure
 
-By default this command assumes the latest Hive master CI build, and the latest OpenShift stable release. `--hive-image` can be specified to use a specific Hive image to run the install, and `--release-image` can be specified to control which OpenShift release image to install in the cluster.
+Credentials will be read from ~/.azure/osServicePrincipal.json typically created via the `az login` command.
+
+```bash
+bin/hiveutil create-cluster --base-domain=azure.mydomain.example.com --cloud=azure mycluster
+```
+
+#### Create Cluster on GCP
+
+Credentials will be read from ~/.gcp/osServiceAccount.json, this can be created by:
+
+ 1. Login to GCP console at https://console.cloud.google.com/
+ 1. Create a service account with the owner role.
+ 1. Create a key for the service account.
+ 1. Select JSON for the key type.
+ 1. Download resulting JSON file and save to ~/.gcp/osServiceAccount.json.
+
+```bash
+bin/hiveutil create-cluster --base-domain=gcp.mydomain.example.com --cloud=gcp --gcp-project-id=myproject mycluster
```

Comment:
We should include text about setting the release image since azure and gcp are still in developer preview and as such not supported in the latest stable OpenShift release.

Reply:
This will get the latest nightly 4.2 release for testing azure and gcp installs.
```
curl -s https://openshift-release.svc.ci.openshift.org/api/v1/releasestream/4.2.0-0.nightly/latest | jq -r '.pullSpec'
```

Reply:
I tried using that URL with --release-image-source but it requires auth, and not the standard registry pull-secret.

I've been testing with registry.svc.ci.openshift.org/origin/release:4.2 however which for whatever reason, does not require any special auth. I will use this for the examples.

Reply:
Sure. The one that I linked is the nightly, which requires creds from the CI cluster. The one that you are using is the developer preview release. Yours is better to reference in the docs.